eng/pipelines/corefx-base.yml

```diff
@@ -139,11 +139,11 @@ jobs:
 
           - ${{ if eq(job.customBuildSteps[0], '') }}:
             
-            - ${{ if eq(parameters.isOfficialBuild, 'true') }}:
+            - ${{ if and(eq(parameters.isOfficialBuild, 'true'), eq(parameters.targetOS, 'Windows_NT')) }}:
```

Comment:
Why do we want to couple restoring internals tools to a specific OS? It would make more sense to condition relevant PackageReferences for Windows_NT. Thinking about ILC which is/was also used for netcoreappaot (windows, unix, osx)

Reply:
Eric asked to restore optional tools for all windows legs. So that is why I thought on using this condition to avoid showing the extra steps in all the other builds. However I can manage to do it through an extra property through the Job. Will fix it.

Reply:
thanks!

Reply:
What about always running it for official builds and letting the project itself decide when it is needed?  Is there any problem with that?

Reply:
The optional tools project? How would we state if it is needed or not? Based on targetgroup and OS?

Reply:
TargetGroup and TargetOS should suffice, yeah.

Reply:
However IBC Merge will always be restored, since that is directly checked in into arcade. Unless we clear all packageReferences whenever it doesn't apply to that TargetGroup/OS.

Reply:
They have conditions on them for exactly that reason: https://github.com/dotnet/arcade/blob/master/eng/common/internal/Tools.csproj#L11-L12

Reply:
I see, so we can just set those properties on Versions.props file based on targetgroup and targetos: 

https://github.com/dotnet/roslyn/blob/2ddd9f252ad27b8aa17e5e934cc2bd04c21a73af/eng/Versions.props#L285

Reply:
right. note that you want to set that directly in InternalTools.props (thanks to property evaluation comes before item evaluation) as we don't import anything else.

Reply:
We need to double check that we're actually passing TargetGroup down to this project.

Reply:
Yes, we would need to pass down TargetGroup to this project when doing the restore in official builds. Locally, I'm not sure if that is happening, I think it should, since we control that directly in Tools.props -- so we should just probably pass it down.

Reply:
No as the Tools.csproj restore is an out-of-proc operation: https://github.com/dotnet/corefx/blob/master/eng/Tools.props#L72

Reply:
But we could potentially pass that value down?

Reply:
Sure. Just saying that it doesn't flow yet.

Reply:
I will be setting UsingToolIbcOptimization in version.props in my changes to enable IBCMerge.

Reply:
Hmm I didn't read your comment, I just set it in OptionalTools.props -- could you please take a look and see if that is enough?